python/pylibcudf/pylibcudf/interop.pyx

```diff
@@ -315,3 +322,77 @@ def _to_arrow_scalar(cudf_object, metadata=None):
     # Note that metadata for scalars is primarily important for preserving
     # information on nested types since names are otherwise irrelevant.
     return to_arrow(Column.from_scalar(cudf_object, 1), metadata=metadata)[0]
+
+
+cpdef Table from_dlpack(managed_tensor):
+    """
+    Convert a DLPack DLTensor into a cudf table.
+
+    For details, see :cpp:func:`cudf::from_dlpack`
+
+    Parameters
+    ----------
+    managed_tensor : PyCapsule
+        A 1D or 2D column-major (Fortran order) tensor.
+
+    Returns
+    -------
+    Table
+        Table with a copy of the tensor data.
+    """
+    cdef unique_ptr[table] c_result
+    cdef DLManagedTensor* dlpack_tensor = <DLManagedTensor*>PyCapsule_GetPointer(
+        managed_tensor, "dltensor"
+    )
+    PyCapsule_SetName(managed_tensor, "used_dltensor")
+
+    with nogil:
+        c_result = move(
+            cpp_from_dlpack(dlpack_tensor)
+        )
+
+    result = Table.from_libcudf(move(c_result))
+    dlpack_tensor.deleter(dlpack_tensor)
```

Comment:
Can we add a comment indicating that our `from_dlpack` implementation always copies, hence this eager deletion? And maybe link to https://github.com/rapidsai/cudf/issues/10874 and https://github.com/rapidsai/cudf/issues/10849 as TODOs? Without that context this implementation looks very odd with the eager deletion.